src/Facets/SquidFacet.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { ISquidRouter } from "../Interfaces/ISquidRouter.sol";
+import { ISquidMulticall } from "../Interfaces/ISquidMulticall.sol";
+import { LibAsset, IERC20 } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { LibUtil } from "../Libraries/LibUtil.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+
+/// @title Squid Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through Squid Router
+contract SquidFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    enum SquidCallType {
+        BridgeCall,
+        CallBridge,
+        CallBridgeCall
+    }
+
+    struct SquidData {
+        SquidCallType callType;
+        bytes callData;
+    }
+
+    /// State ///
+    ISquidRouter public immutable squidRouter;
+    ISquidMulticall public immutable squidMulticall;
+
+    /// Constructor ///
+    constructor(ISquidRouter _squidRouter, ISquidMulticall _squidMulticall) {
+        squidRouter = _squidRouter;
+        squidMulticall = _squidMulticall;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Squid Router
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _squidData data specific to Squid Router
+    function startBridgeTokensViaSquid(
+        ILiFi.BridgeData memory _bridgeData,
+        SquidData memory _squidData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
```

Comment:
This is not really true, as we may do both in squid, so the check should be based on that too, comparing the `_bridgeData` flag with the parameters passed in `_squidData`.